mods/mc_teacher/gui_dash.lua

```diff
@@ -659,414 +657,348 @@ end
+                    minetest.log("warning", "mc_teacher gui_dash.lua: realm with ID: " .. mdata.realm_id[context.selected] .. " does not exist but is associated with a classroom.")
+                end
+            else
+                minetest.chat_send_player(pname, pname .. ": Please click on a classroom in the list to join.")
+            end
+        elseif fields.back then
+            show_teacher_menu(player)
+        elseif fields.delete then
+
+            if context.selected then
+                pmeta = player:get_meta()
+                pdata = minetest.deserialize(pmeta:get_string("classrooms"))
+
+
+                -- Update modstorage first
+                mdata = minetest.deserialize(minetest_classroom.classrooms:get_string("classrooms"))
+                loc = check_access_code(pdata.access_code[context.selected], mdata.access_code)
+
+                --Delete the realm
+                local realm = Realm.realmDict[mdata.realm_id[loc]]
+                if (realm ~= nil) then
+                    realm:Delete()
+                end
+
+                mdata.course_code[loc] = nil
+                mdata.section_number[loc] = nil
+                mdata.start_year[loc] = nil
+                mdata.start_month[loc] = nil
+                mdata.start_day[loc] = nil
+                mdata.end_year[loc] = nil
+                mdata.end_month[loc] = nil
+                mdata.end_day[loc] = nil
+                mdata.classroom_map[loc] = nil
+                mdata.access_code[loc] = nil
+                mdata.realm_id[loc] = nil
+                minetest_classroom.classrooms:set_string("classrooms", minetest.serialize(mdata
+                ))
+
+                -- Update player metadata
+                pdata.course_code[context.selected] = nil
+                pdata.section_number[context.selected] = nil
+                pdata.start_year[context.selected] = nil
+                pdata.start_month[context.selected] = nil
+                pdata.start_day[context.selected] = nil
+                pdata.end_year[context.selected] = nil
+                pdata.end_month[context.selected] = nil
+                pdata.end_day[context.selected] = nil
+                pdata.classroom_map[context.selected] = nil
+                pdata.access_code[context.selected] = nil
+                pmeta:set_string("classrooms", minetest.serialize(pdata
+                ))
+
+                show_classrooms(player)
+            else
+                minetest.chat_send_player(pname, pname .. ": Please click on a classroom in the list to delete.")
+            end
+
+            -- TODO: eventually delete this button, only useful for testing
+        elseif fields.deleteall then
+            pmeta:set_string("classrooms", nil)
+            minetest_classroom.classrooms:set_string("classrooms", nil)
+            show_classrooms(player)
+        else
+            -- escape without input
+            return
         end
     end
```

Comment:
The code for handling the plant compendium formspecs (from `magnify`) also seems to have been lost while resolving merge conflicts

```suggestion
    end
    
    if formname == "mc_teacher:species_menu" then
		if fields.back then
		  	show_teacher_menu(player)
		elseif fields.species_list then
        	local event = minetest.explode_textlist_event(fields.species_list)
        	if event.type == "CHG" then
        		context.species_selected = event.index
        	end
		elseif fields.condensed_view or fields.expanded_view then
			if context.species_selected then
      			local ref = get_species_ref(context.species_selected)
          		local full_info = magnify.get_species_from_ref(ref)

				if full_info ~= nil then
          			if fields.condensed_view then -- condensed
						minetest.show_formspec(pname, "mc_teacher:species_condensed", get_condensed_species_formspec(full_info.data))
            		else -- expanded
            			minetest.show_formspec(pname, "mc_teacher:species_expanded", get_expanded_species_formspec(full_info.data, full_info.nodes, ref))
            		end
				else
					minetest.chat_send_player(pname, "An entry for this species exists, but could not be found in the plant database.\nPlease check your server's plant database files to ensure all plants were registered properly.")
				end	
        	end
		end
	end
	
	if formname == "mc_teacher:species_condensed" then
      	if fields.back then 
        	show_species(player)
	    end
    end
    
    if formname == "mc_teacher:species_expanded" then
      	if fields.back then
          	show_species(player)
      	end
    end
```